### PR TITLE
Feature/handle progress cancel

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/LtsApiSensorContext.java [hunk 1]
```diff
@@ -105,8 +106,8 @@ public boolean matchFilePattern(String antPattern) {
   private final Project project;
 
   public LtsApiSensorContext(Project project, InputModule inputModule, Settings settings, FileSystem fs, ActiveRules activeRules, SensorStorage sensorStorage,
```
Comment (SonarTech) [inline core/src/main/java/org/sonarsource/sonarlint/core/analyzer/sensor/LtsApiSensorContext.java]: MAJOR Constructor has 8 parameters, which is greater than 7 authorized. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00107)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: core/src/main/java/org/sonarsource/sonarlint/core/StandaloneSonarLintEngineImpl.java [hunk 2]
```diff
@@ -80,20 +80,15 @@ public RuleDetails getRuleDetails(String ruleKey) {
   }
 
   @Override
-  public AnalysisResults analyze(StandaloneAnalysisConfiguration configuration, IssueListener issueListener) {
-    return analyze(configuration, issueListener, null);
```
Comment (dbmeneses) [inline core/src/main/java/org/sonarsource/sonarlint/core/StandaloneSonarLintEngineImpl.java]: removing it to discourage not specifying a progress? for tests this was useful.

Comment (henryju) [inline core/src/main/java/org/sonarsource/sonarlint/core/StandaloneSonarLintEngineImpl.java]: This is mainly to keep a simple interface, without having multiple overloaded methods. Also it forces clients to update and decide what to do.

**Pull request merged**